### PR TITLE
✨(app) configure a basic LTI configuration icon

### LINKED ISSUE #88: Offer an LTI icon for Warren external tool
## Feature Request

LMS include a feature to render an icon for an external tool. This visual representation simplifies navigation and enhances the user experience, making it easier for educators, administrators, and students to access Warren's educational tools within the system.

- [x] Design a basic Icon that represents analytics
- [x] Add it to the Django static files
- [x] Set its URL properly in the LTI configuration XML file.

### PR DESCRIPTION
**Pull request by lebaudantoine**
Integrate a basic configuration icon into the interface, with plans to refine its design later.
It closes #88.

Capture d’écran 2023-10-04 à 01 58 59

### CODE REVIEW

File: src/app/apps/lti/views.py [hunk 1]
```diff
@@ -136,14 +137,20 @@ class LTIConfigView(TemplateView):
 
     def get_context_data(self, **kwargs):
         """Get context data for rendering the template."""
+        host = self.request.get_host()
+        icon_url = (
+            f"//{host}{static(settings.LTI_CONFIG_ICON)}"
+            if settings.LTI_CONFIG_ICON
+            else ""
+        )
```
Comment (jmaupetit) [inline src/app/apps/lti/views.py:145]:
```suggestion
        icon_url = (
            self.request.build_absolute_uri(static(settings.LTI_CONFIG_ICON))
            if settings.LTI_CONFIG_ICON
            else ""
        )
```

File: src/app/apps/lti/views.py [hunk 2]
```diff
@@ -136,14 +137,20 @@ class LTIConfigView(TemplateView):
...
         return {
             "code": settings.LTI_CONFIG_TITLE.lower()
             if settings.LTI_CONFIG_TITLE
             else None,
             "contact_email": settings.LTI_CONFIG_CONTACT_EMAIL,
             "description": settings.LTI_CONFIG_DESCRIPTION,
-            "host": self.request.get_host(),
-            "icon_url": settings.LTI_CONFIG_ICON,
+            "host": host,
+            "icon_url": icon_url,
```
Comment (jmaupetit) [inline src/app/apps/lti/views.py:153]:
```suggestion
            "host": self.request.get_host(),
            "icon_url": icon_url,
```

File: src/app/staticfiles/warren_32x32.png
Comment (jmaupetit) [inline src/app/staticfiles/warren_32x32.png:1]: I've read from the [old LTI 1.0 spec](https://www.imsglobal.org/specs/lticiv1p0/specification) that the icon recommended size is 50x50 px. I did not find any other size recommendation from recent LTI versions. Do you think we must switch to SVG?

**Review by jmaupetit — COMMENTED** [review]
Well done!

File: src/app/staticfiles/warren_32x32.png
Comment (lebaudantoine) [reply src/app/staticfiles/warren_32x32.png:1]: Done!

File: src/app/apps/lti/views.py (hunk 1 above)
Comment (lebaudantoine) [reply src/app/apps/lti/views.py:145]: Actually, the config takes two schemes, one secure and one insecure. The `icon_url` is misleading, as we expect a subpart of the URL, the domain and the path to the icon file.

Comment (lebaudantoine) [reply src/app/apps/lti/views.py:145]: I have renamed the variable, wdyt?

**Review by jmaupetit — APPROVED** [review]

**Pull request merged**